File: hs/codewars/sum_of_the_first_nth_term_of_series.hs
-- https://www.codewars.com/kata/sum-of-the-first-nth-term-of-series

module Codewars.Kata.NthSeries where

import Prelude
import Text.Printf

seriesSum :: Integer -> String
seriesSum n = printf "%.2f" (foldr (+) 0.0 $
  [1.0 / (fromIntegral $ 1 + 3 * x) | x <- [0..(n-1)]] :: Double)

File: clojure/codewars/sum_of_the_first_nth_term_of_series.clj
; https://www.codewars.com/kata/sum-of-the-first-nth-term-of-series

(ns nthseries.core)

(defn series-sum [n]
  (->> (range n)
       (map #(/ (inc (* 3 %))))
       (reduce + 0.0)
       (format "%.2f")))
